src/main/java/com/google/devtools/build/lib/remote/util/DigestOutputStream.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 The Bazel Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.devtools.build.lib.remote.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import build.bazel.remote.execution.v2.Digest;
+import com.google.common.annotations.Beta;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hasher;
+import java.io.FilterOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * An {@link OutputStream} that maintains a {@link Digest} of the data written to it.
+ *
+ * <p>Based on Guava's {@link com.google.common.hash.HashingOutputStream}.
```

Comment:
s/Based on/Similar to/ and also add that it returns a Digest object that's specific to the remote execution protocol.

Reply:
done